dev/com.ibm.ws.security.audit.source/src/com/ibm/ws/security/audit/encryption/AuditCrypto.java

```diff
@@ -1196,20 +1214,31 @@ static final byte[] signISO9796(byte[][] key, byte[] data, int off, int len, boo
         }
 
         byte[] sig = null;
-        if (useJCE) {
+        if (fips140_3Enabled || useJCE) {
             try {
                 BigInteger n = new BigInteger(key[0]);
                 BigInteger e = new BigInteger(key[2]);
                 BigInteger p = new BigInteger(key[3]);
                 BigInteger q = new BigInteger(key[4]);
                 BigInteger d = e.modInverse((p.subtract(BigInteger.ONE)).multiply(q.subtract(BigInteger.ONE)));
                 KeyFactory kFact = null;
-                kFact = KeyFactory.getInstance(CRYPTO_ALGORITHM);
-
+                if (fips140_3Enabled)
+                    kFact = KeyFactory.getInstance(CryptoUtils.CRYPTO_ALGORITHM_RSA, CryptoUtils.IBMJCE_PLUS_FIPS_NAME);
```

Comment:
```suggestion
                    kFact = KeyFactory.getInstance(CryptoUtils.CRYPTO_ALGORITHM_RSA, provider);
```

Is this right?  For Semeru 17 wouldn't it be open jce plus fips?  It looks like `provider` is passed in other places instead of `CryptoUtils.IBMJCE_PLUS_FIPS_NAME`.

Reply:
We have not start Semeru 17 yet. We should get the right provider for Semuru too so I fixed it.